### ISSUE TITLE
Possible bug - Multiplying dropped features per Convolution Block

### ISSUE THREAD

**Issue by iliaschalkidis** [tensorflow/tensor2tensor#1356]
I think I found out a possible error in the implementation of the stack of convolution blocks with residual connection, which are being used both in ByteNet Encoder and Decoder.

The current implementation is as follows:

```
def residual_dilated_conv(x, repeat, padding, name, hparams):
  """A stack of convolution blocks with residual connections."""
  with tf.variable_scope(name):
    k = (hparams.kernel_height, hparams.kernel_width)
    dilations_and_kernels = [((2**i, 1), k)
                             for i in range(hparams.num_hidden_layers)]
    for i in range(repeat):
      with tf.variable_scope("repeat_%d" % i):
        y = common_layers.conv_block(
            common_layers.layer_norm(x, hparams.hidden_size, name="lnorm"),
            hparams.hidden_size,
            dilations_and_kernels,
            padding=padding,
            name="residual_conv")
        y = tf.nn.dropout(y, 1.0 - hparams.dropout)
        x += y
    return x
```

I suggest that lines 45-46 should be inverted. This change can prevent possible errors without alternating the final outcome.

https://github.com/tensorflow/tensor2tensor/blob/abca2107d0359483bc925dddf13efea6eeb3b2ab/tensor2tensor/models/bytenet.py#L45

The reason behind this observation is that given the current order of these two operations, the ratio of the dropped out features will be amplified by each repetition.

For example:

Let's say that `x` (inputs) have been already passed by a dropout layer and some features (dimensions) have already been zeroed out. Then the outputs `y` of `conv_block` are also dropped (line 45), where possibly a different set of dimensions have been zeroed out. Then in the following addition `x += y` (line 46), features (dimensions) that have been zeroed out in both `x` and  `y` will pass over in the updated version of `x`. Given multiple repetitions (`repeat`), the features (dimensions) that have been zeroed out will be amplified.

```
x[0] = [0.1, 0.0, 0.0, 0.2]
y[0] = conv_block(x)[0] = [0.4, 0.2, 0.3, 0.1]
y[0] = dropout(y)[0] = [0.4, 0.2, 0.3, 0.0]
x[0] = (x += y)[0] = [0.5, 0.0, 0.0, 0.0]
......
x[0] = [0.0, 0.0, 0.0, 0.0]
```
